### ISSUE TITLE
Automatic reconnect to disconnected multiplayer games (GNA #18158)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1261]
**[Original submission](https://gna.org/bugs/index.php?18158)** by whitemage on 2011-05-21

As I remember, about a year ago when I was disconnected from an official multiplayer game and I reconnected to the server with my ID and password then I automatically joined the game from where I was disconnected from. Not any more. Now I painfully have to search for the game I was disconnected from (if I remember the name and I am fast enough). To make the matter worse I also have to watch back the whole game like a movie since I am an observer at that point. This shows the opponent�s moves (which it shouldn�t) and wastes time lowering my chances to be able to get back. 

Recommendation: make automatic join and automatic reassignment to multiplayer games on the official server directly to the point where the player was disconnected from. Also, give a grace period of 90 seconds for disconnected players to rejoin so other players will know about this and will be denied from moving the player�s units. 90 seconds should be enough for most players to rejoin if they want and they should not be penalized. For example see the attached recorded game where it would have helped me. I used my userID and password to join the official server.

(Reproduced on any)
Release: 1.8.5 and 1.8.6
Priority: 5 - Normal
Severity: 1 - Wish

**Comment by wesnoth-bugs** [comment]
**Modified on 2011-05-21**

**whitemage added attachment:** [4p_-_Isar's_Cross_Turn_16_(1325).gz](http://gna.org/bugs/download.php?file_id=13034)

**Comment by wesnoth-bugs** [comment]
**Modified on 2011-05-22**

**soliton changed severity:** 4 - Important -> 1 - Wish

**soliton wrote:**

> There's never been an automatic rejoin function.
> 
> 
> 
> You can check the "Quick Replays" option in the lobby to get faster through the replay.
> 
> 
> 
> There is no way to know whether a disconnected player comes back at all so we're not going to force others to wait if they don't want to. What is possible is to give a better message if the disconnect is because of a reconnect and subsequent kick of the old connection though.

**Comment by Wedge009** [comment]
Seems strongly related to #1034, although this one wants to make the reconnection automatic.

**Comment by alasiax** [comment]
I haven't been able to find a way to rejoin a game other than as an observer, but which time my friend has already had the dialogue to chose AI or take over, and it doesn't list me as an option even after I'm observing. (If this is inaccurate, or there's a way to rejoin a game without saving and recommencing the server, please feel free to chime in!)

This feature would be a great addition to MP games. The option for the server to wait for a disconnected player to rejoin would be also be welcome (per #1034), as it would allow for patience rather than the save restart dance that currently occurs. 

Usage case: I'm playing with a friend I know, and have means to contact them outside the game (we're often on the phone while we're playing, or texts are sent frequently). My friend knows I'll be back online shortly, so it's not a problem to wait for me to reconnect.

**Comment by soliton-** [comment]
Yes, you can only rejoin as observer but you don't need to save and restart. The remaining player is (or should become) the host and can hand you control back once you joined as an observer. See the `control` or `give_control` command on https://wiki.wesnoth.org/CommandMode.

(Yes, this feature is still a good idea.)
